### PR TITLE
Preparing for 0.52.0rc3

### LINKED ISSUE #6443: Pin llvmlite for RC3... Final of 0.52.0
Reminder to do that ^

### PR DESCRIPTION
**Pull request by sklam**
cherry picked PRs and update changelog.
~Still pending https://github.com/numba/numba/issues/6443~
closes #6443
pending #6480

### CODE REVIEW

**Comment by sklam** [comment]
Pinned llvmlite closes #6443

**Comment by sklam** [comment]
oops, shouldn't be pinning to 0.36, should be 0.35

File: numba/core/callconv.py [hunk 1]
```diff
@@ -543,9 +543,16 @@ def _get_return_argument(self, func):
...
         return func.args[1]
 
-    def call_function(self, builder, callee, resty, argtys, args):
+    def call_function(self, builder, callee, resty, argtys, args,
+                      attrs=None):
         """
         Call the Numba-compiled *callee*.
+        Parameters:
+        -----------
+        attrs: LLVM style string or iterable of individual attributes, default
```
Comment (stuartarchibald) [inline numba/core/callconv.py:552]: Just noticed this docstring is wrong!

**Comment by stuartarchibald** [comment]
Changes up to 20bd425 look good.

File: numba/core/callconv.py (hunk 1 above)
Comment (sklam) [reply numba/core/callconv.py:552]: heh, fix it in final

**Comment by stuartarchibald** [comment]
16af7f9  looks good.

**Review by stuartarchibald — APPROVED** [review]

**Pull request merged**